compiler/test/fail_compilation/test20859.d

```diff
@@ -0,0 +1,11 @@
+/**
+TEST_OUTPUT:
+---
+fail_compilation/test20859.d(8): Error: variable `test20859.ICE.__vtbl` conflicts with variable `test20859.ICE.__vtbl` at fail_compilation/test20859.d(10)
```

Comment:
can we give a better error message here?

Reply:
In general, there should be an error for using identifiers starting with `__` because they are reserved, but so far I haven't found a good place to do that since there isn't always a clear distinction between 'generated/druntime' code and user code. So I think it's a more involved problem than simply fixing this ICE.